src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -598,6 +625,25 @@ public static unsafe object InitializeCustomMarshaller(RuntimeTypeHandle pParame
             return marshaller;
         }
 
+        [ThreadStatic]
```

Comment:
Can you put these under the ifdef as well (static fields don't treeshake well)?

Reply:
Now in a separate file, that is only compiled for macOS.